src/Processors/Executors/ExecutorTasks.cpp

```diff
@@ -32,14 +32,17 @@ void ExecutorTasks::tryWakeUpAnyOtherThreadWithTasks(ExecutionThreadContext & se
 {
     if (!task_queue.empty() && !threads_queue.empty() && !finished)
     {
-        size_t next_thread = self.thread_number + 1 == num_threads ? 0 : (self.thread_number + 1);
+        size_t next_thread = (self.thread_number + 1) % use_threads;
```

Comment:
I would be happier to avoid the division (especially if for some reason use_threads = 0).
But probably it's ok